#170045960 build the signup endpoint with database

Nigorjeanluc opened the pull request:
### What does this PR do?
This pull request set up the route to create new user's account and save data in a database

### Description of Task to be completed?
The tasks done here are
- add validation to this endpoint for v2
- add models and integrate DB to save user's data
- add a method for the API to handle this endpoint

### How should this be manually tested?
Clone or download the develop branch to your local machine and you can see the changes.

### What are the relevant pivotal tracker stories?
[#170045960](https://www.pivotaltracker.com/n/projects/2410710/stories/170045960)

Nigorjeanluc committed:
ft(signup-endpoint): taks done here are
- add validation to this endpoint for v2
- add models and integrate db to save user's datat
- add a method for the api to handle this endpoint
[Finishes #170045960]

mwibutsa commented on server/v2/controllers/userController.js:
```diff
@@ -0,0 +1,65 @@
+                });
+            }
+
+            const existingUsername = await pool.query(queries.findOneUsername, [req.body.username]);
```
Move database related operations to the relevant Model; for instance, the methods to create, update, delete and get user info should all be located in user Model.

mwibutsa commented on server/v2/tests/auth.spec.js:
```diff
@@ -0,0 +1,62 @@
+                res.body.should.be.a("object");
+                done();
+            });
+    });
```
As we have all agreed you should be using async-await in your tests. it is the Modern way of writing tests and is preferred by most developers than using callbacks.

mwibutsa commented on server/v2/tests/auth.spec.js:
```diff
@@ -0,0 +1,62 @@
+                );
+                done();
+            });
+    });
```
This also needs to use async await

mwibutsa commented:
if there is a way you can keep your tests coverage to 100% it would be better but make sure you don't go bellow 95%

Nigorjeanluc committed:
ft(signup-endpoint): LFA's feedback on this endpoint
- modify tests from callbacks to promises
- put all queries in controller into the model
[Finishes #170045960]

mwibutsa commented on server/v2/tests/auth.spec.js:
```diff
@@ -0,0 +1,51 @@
+            expect(res.body.status).to.equal(201);
+            expect(res.body).to.be.an('object');
+            expect(res.body.message).to.be.a('string');
+        });
```
I'm glad that you are using async-await in your tests. Usually, when you are using async-await you need to nest the block inside a try-catch block to capture the errors.  that's the only thing which is missing in your test.

Nigorjeanluc replied:
OK, I'm going to work on it according to the resource you gave us yesterday. Thanks again

mwibutsa reviewed (changes requested):
Even though you are writing tests you need to handle the case in which there might be an error. consider using nesting any async await code inside a ```try catch```  block.

mwibutsa commented:
@Nigorjeanluc, Don't forget to squash your commits into one. it looks more professional when they are squashed.

Nigorjeanluc committed:
ft(signup-endpoint): LFA's feedback on this endpoint
- modify integrate try/catch block into my test's promises
[Finishes #170045960]

mwibutsa reviewed (approved):
This work looks amazing @Nigorjeanluc
